src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -31,6 +32,24 @@ class AutowirePass extends AbstractRecursivePass
     private $ambiguousServiceTypes = array();
     private $autowired = array();
     private $lastFailure;
+    private $throwOnAutowiringException;
+    private $autowiringExceptions = array();
```

Comment:
Oh, except that I'm clearing it at the *beginning* of the process. The pass *does* need persistent state - the other pass uses it. There's not a "better" way of doing this (i.e. storing somewhere in the container) that I'm aware of. Anyways, I *am* now clearing it at the beginning of process, so if it's called twice, it's cleared between.